examples/sadtalker/datasets/dataset_audio2coeff.py

```diff
@@ -0,0 +1,270 @@
+import os
+import random
+from tqdm import tqdm
+import scipy.io as scio
+import numpy as np
+import mindspore as ms
+import utils.audio as audio
+
+
+def crop_pad_audio(wav, audio_length):
+    if len(wav) > audio_length:
+        wav = wav[:audio_length]
+    elif len(wav) < audio_length:
+        wav = np.pad(wav, [0, audio_length - len(wav)], mode="constant", constant_values=0)
+    return wav
+
+
+def parse_audio_length(audio_length, sr, fps):
+    bit_per_frames = sr / fps
+
+    num_frames = int(audio_length / bit_per_frames)
+    audio_length = int(num_frames * bit_per_frames)
+
+    return audio_length, num_frames
+
+
+def generate_blink_seq(num_frames):
+    ratio = np.zeros((num_frames, 1))
+    frame_id = 0
+    while frame_id in range(num_frames):
+        start = 80
+        if frame_id + start + 9 <= num_frames - 1:
+            ratio[frame_id + start : frame_id + start + 9, 0] = [
+                0.5,
+                0.6,
+                0.7,
+                0.9,
+                1,
+                0.9,
+                0.7,
+                0.6,
+                0.5,
+            ]
+            frame_id = frame_id + start + 9
+        else:
+            break
+    return ratio
+
+
+def generate_blink_seq_randomly(num_frames):
+    ratio = np.zeros((num_frames, 1))
+    if num_frames <= 20:
+        return ratio
+    frame_id = 0
+    while frame_id in range(num_frames):
+        start = random.choice(range(min(10, num_frames), min(int(num_frames / 2), 70)))
+        if frame_id + start + 5 <= num_frames - 1:
+            ratio[frame_id + start : frame_id + start + 5, 0] = [
+                0.5,
+                0.9,
+                1.0,
+                0.9,
+                0.5,
+            ]
+            frame_id = frame_id + start + 5
+        else:
+            break
+    return ratio
+
+
+def read_filelist(input_path):
+    audios = []
+    images = []
+
+    if os.path.isfile(input_path):
+        ext = input_path.split(".")[-1]
+        if ext == "txt":
+            with open(input_path, "r") as f:
+                for line in f.read().splitlines():
+                    audio_path, first_image_path = line.split(" ")
+
+                    if os.path.isfile(audio_path) and os.path.isfile(first_image_path):
+                        audios.append(audio_path)
+                        images.append(first_image_path)
+
+    return audios, images
+
+
+class TestDataset:
```

Comment:
name update for`TestDataset`

Reply:
Thanks for the suggestion. This part has been refactored, please take a look if you have time~